File: project/seryeonha-onboarding/src/main/java/com/icd/onboarding/survey/repository/SurveyRepository.java
package com.icd.onboarding.survey.repository;

import com.icd.onboarding.survey.domain.Survey;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface SurveyRepository extends JpaRepository<Survey, Long> {
}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/repository/SurveyRepository.kt
package com.wlghsp.querybox.repository

import com.wlghsp.querybox.domain.survey.Survey
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.Param

interface SurveyRepository : JpaRepository<Survey, Long> {

    @EntityGraph(attributePaths = ["questions.values"])
    fun findSurveyWithQuestionsById(@Param("id") id: Long): Survey?
}